contracts/secret-storage/SecretManager.sol

```diff
@@ -0,0 +1,588 @@
+        uint256 remainingDeposit = userStorage[_secretId].usdcDeposit;
+        delete userStorage[_secretId];
+        USDC_TOKEN.safeTransfer(owner, remainingDeposit);
+    }
+
+    function _updateSecretEndTimestamp(
+        uint256 _secretId,
+        uint256 _endTimestamp,
+        uint256 _usdcDeposit,
+        address _owner
+    ) internal {
+        if(userStorage[_secretId].owner != _owner)
+            revert SecretManagerNotUserStoreOwner();
+
+        if(_endTimestamp < block.timestamp)
+            revert SecretManagerInvalidEndTimestamp();
+
+        uint256 currentEndTimestamp = userStorage[_secretId].endTimestamp;
+        if(block.timestamp > currentEndTimestamp)
+            revert SecretManagerAlreadyTerminated();
+
+        if(_endTimestamp > currentEndTimestamp) {
+            USDC_TOKEN.safeTransferFrom(_owner, address(this), _usdcDeposit);
+            userStorage[_secretId].usdcDeposit += _usdcDeposit;
+
+            uint256 minUsdcDeposit = (_endTimestamp - userStorage[_secretId].startTimestamp) * userStorage[_secretId].sizeLimit * SECRET_STORE_FEE_RATE * NO_OF_NODES_TO_SELECT;
+            _checkUsdcDeposit(userStorage[_secretId].usdcDeposit, minUsdcDeposit);
+        }
+        else {
+            uint256 removedDuration = currentEndTimestamp - _endTimestamp;
+            uint256 usdcRefund = removedDuration * userStorage[_secretId].sizeLimit * SECRET_STORE_FEE_RATE * NO_OF_NODES_TO_SELECT;
+            
+            userStorage[_secretId].usdcDeposit -= usdcRefund;
+            USDC_TOKEN.safeTransfer(_owner, usdcRefund);
+        }
+
+        userStorage[_secretId].endTimestamp = _endTimestamp;
+
+        emit SecretEndTimestampUpdated(_secretId, _endTimestamp);
+    }
+
+    function _terminateSecret(
+        uint256 _secretId,
+        address _owner
+    ) internal {
+        _updateSecretEndTimestamp(_secretId, block.timestamp, 0, _owner);
+
+        emit SecretTerminated(_secretId);
+    }
+
+    //-------------------------------- internal functions end ----------------------------------//
+
+    //------------------------------- external functions start ----------------------------------//
+
+    function createSecret(
+        uint256 _sizeLimit,
+        uint256 _endTimestamp,
+        uint256 _usdcDeposit
+    ) external {
+        _createSecret(_sizeLimit, _endTimestamp, _usdcDeposit, _msgSender());
+    }
+
+    function acknowledgeStore(
+        uint256 _secretId,
+        uint256 _signTimestamp,
+        bytes memory _signature
+    ) external {
+        _acknowledgeStore(_secretId, _signTimestamp, _signature);
+    }
+
+    function acknowledgeStoreFailed(
+        uint256 _secretId
+    ) external {
+        _acknowledgeStoreFailed(_secretId);
+    }
+
+    function markStoreAlive(
+        uint256 _secretId,
+        uint256 _signTimestamp,
+        bytes memory _signature
+    ) external {
+        _markStoreAlive(_secretId, _signTimestamp, _signature, _msgSender());
```

Comment:
this transaction can be sent by anyone, hence don;t use msgSender here. And get the enclave owner from registered enclave list on secret store contract